### DISCUSSION TITLE
How to develop a BRC20 tool application ?

### DISCUSSION THREAD

**Discussion by vccomnet** [trustwallet/wallet-core#3567] [Q&A]
I am learning to use wallet core and want to develop a BRC20 tool software. I have looked at the sample code in the tests section, including: TEST (Bitcoin Signing, SignBRC20TransferCommit), TEST (Bitcoin Signing, SignBRC20TransferReveal), and TEST (Bitcoin Signing, SignBRC20TransferInsertion), but I cannot see what their respective functions are, which is deploy, which is mint, and which is transfer, The example does not have a method to set op. Please help explain, thank you.

**Comment by lamafab** [comment]
Hey vccomnet

We added the Deploy and Mint methods during development, but they were removed in the final PR since the Transfer method was the top priority and we lacked tests for the former two. In other words; as of now we only support Transfer. I do not know yet when/if Deploy and Mint will be implemented.

I hope [the graph](https://github.com/trustwallet/wallet-core/pull/3233#issuecomment-1601003573) and the tests that you saw make it clear on how the API is used?

**Comment by weixuefeng** [comment]
@lamafab hi，First of all, thank you for your work on this. I used `deploy` on the previous version of wallet-core, and this version deleted the relevant code. I'm wondering if there is some security risk here? Or is it simply due to not enough testing? (mayble i only support IMAGE/JPEG TYPE)

**Reply by lamafab** [reply]
Hey, there's no security risk. Implementing deploy/mint is very easy, almost identical to transfer. We basically removed it because we had no tests. However, I'm currently working on something else right now, but we can probably add this back, soon.

**Reply by weixuefeng** [reply]
ok. thanks for your reply. i will try to complete it and make a pull request.
